content/Certification/certifying_a_product/self_pre_test.md

```diff
@@ -0,0 +1,77 @@
++++
+title = "Self Pre-Test"
+chapter = false
+weight = 7
++++
+
+![](../imgs/self_pre_test.png)
+
+Before taking a product for formal certification, doing a self-pre-test is
+important to identify and fix certification-blocking bugs.
+
+During formal certification, tests are run through the test harness, which
+provides logs and certification materials for submission to the certification
+program. In the self-pre-test phase, tests can either be run through the test
+harness or locally on a desktop for faster test iteration.
+
+## Test selection and PICS
+The set of tests required to certify a device is determined from the
+Protocol Implementation Conformance Statement (PICS). This conformance statement
+is filled out by the manufacturer using the [PICS tool](https://picstool.csa-iot.org/)
+and the set of templates for a specific release of the specification:
+[PICS XML files](https://groups.csa-iot.org/wg/members-all/document/folder/2269).
+
+Detailed information about PICS, how to fill them out, and the tools available is
+available in the [PICS guide](https://project-chip.github.io/connectedhomeip-doc/testing/pics_and_pixit.html).
+
+## Running tests
+When tests are run for certification, they are run through the test harness, which is
+used to assist ATLs with test selection and the logs collection required for submissions
+to the certification program. The test harness runs on a raspberry pi, external to the
+development computer. This is the official tool for test submission for certification.
+
+When iterating on software development, it is often easier to run tests locally. It
+is common for manufacturers to run tests locally until they are certain the tests are
+passing, then move on to use the official tool.
+
+The certification program uses a combination of different types of testing for
+certification.
+
+YAML-based tests are common for clusters. These tests are run using an external
+runner to parse and run the YAML code through chip-tool.
+
+Python-based tests use the python controller directly from the chip python library.
```

Comment:
Is this distinction important for this guide?  If so, why?  As a writer of a new test, why would I choose YAML over python?

Reply:
Yup - not that important. Removed. 

The which to pick discussion is covered in the SDK testing guide, but that's also probably shouldn't be part of this guide.

Changed this to omit this information, and also added links to where to find the tests because that seems important if you're planning to run them.